experimental/LieAlgebras/src/serialization.jl

```diff
@@ -0,0 +1,248 @@
+###############################################################################
+#
+#   Lie algebras
+#
+###############################################################################
+
+@register_serialization_type AbstractLieAlgebra uses_id
+
+function save_object(s::SerializerState, L::AbstractLieAlgebra)
+  save_data_dict(s) do
+    save_typed_object(s, coefficient_ring(L), :base_ring)
+    save_object(s, _struct_consts(L), :struct_consts)
+    save_object(s, symbols(L), :symbols)
+    save_attrs(s, L)
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{<:AbstractLieAlgebra})
+  R = load_typed_object(s, :base_ring)
+  struct_consts = load_object(s, Matrix, (sparse_row_type(R), R), :struct_consts)
+  symbs = load_object(s, Vector, Symbol, :symbols)
+  L = lie_algebra(R, struct_consts, symbs; check=false)
+  load_attrs!(s, L)
+  return L
+end
+
+@register_serialization_type LinearLieAlgebra uses_id
+
+function save_object(s::SerializerState, L::LinearLieAlgebra)
+  save_data_dict(s) do
+    save_typed_object(s, coefficient_ring(L), :base_ring)
```

Comment:
we could add a use_params for each LieAlgebra type, then the save_type_params function would take care of the coefficient ring, this would probably also make DirectSumAlgebras files a bit neater.

Reply:
I tried it locally, but I prefer the current form